romeo/src/stacks_client.rs

```diff
@@ -180,15 +180,30 @@ impl StacksClient {
             .unwrap()
     }
 
-    fn nonce_url(&self) -> reqwest::Url {
+    fn cachebust(&self, mut url: reqwest::Url) -> reqwest::Url {
         let mut rng = thread_rng();
         let random_string: String = (0..16).map(|_| rng.sample(Alphanumeric) as char).collect();
 
-        // We need to make sure node returns the uncached nonce, so we add a cachebuster
+        let mut query = url
+            .query()
+            .map(|query| query.to_string())
+            .unwrap_or_default();
+
+        query = if query.is_empty() {
```

Comment:
This is neurotic, but thoughts on something like:

```rust
query = match query.is_empty() {
  true => format!("cachebuster={}", random_string),
  false => format!("{}&cachebuster={}", query, random_string)
}
```

I prefer match when the conditional returns a value.

Reply:
It's not that I don't like it, but considering we're matching on a boolean it doesn't feel idiomatic. Anyways, I'll try it and see if cargofmt keeps it formatted as your suggestion. I do dislike having a lot of open / closing bracket lines.

Reply:
Yep I did what you suggested. It looks nicer.